src/uart.h

```diff
@@ -0,0 +1,150 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/file.h>
+#include <termios.h>
+#include <unistd.h>
+
+#pragma once
+
+#define DEVICE_PATH_SIZE 128
+
+#define APBUART_NUM 6
+#define APBUART_BUFFER_SIZE 1024
+#define APB_START 0x80000000
+
+#define APBUART_FLAG_MASK 0x01
+#define APBUART_REGISTER_TYPE_MASK 0xFF
+#define APBUART_ADDR_MASK 0xFFFF
+#define APBUART_CONTROL_REGISTER_WRITE_MASK 0xEBF
+
+#define APBUART_DATA_REGISTER_ADDRESS 0x00
+#define APBUART_STATUS_REGISTER_ADDRESS 0x04
+#define APBUART_CONTROL_REGISTER_ADDRESS 0x08
+
+#define APBUART0_START_ADDRESS 0x80000100
+#define APBUART1_START_ADDRESS 0x80100100
+#define APBUART2_START_ADDRESS 0x80100200
+#define APBUART3_START_ADDRESS 0x80100300
+#define APBUART4_START_ADDRESS 0x80100400
+#define APBUART5_START_ADDRESS 0x80100500
+
+#define APBUART0_IRQ 2
+#define APBUART1_IRQ 17
+#define APBUART2_IRQ 18
+#define APBUART3_IRQ 19
+#define APBUART4_IRQ 20
+#define APBUART5_IRQ 21
+
+/* Number of simulator ticks between flushing the UARTS.  */
+/* For good performance, keep above 1000.  */
+#define UART_FLUSH_TIME	  5000
+
+/* New uart defines.  */
+#define UART_TX_TIME	1000
+#define UART_RX_TIME	1000
+
+/* Status Register definition, taken from GR712RC documentation:
+    (0)       DR - Data ready: Indicates that new data is available in the receiver holding register.
+    (1)       TS - Transmitter shift register empty: Indicates that the transmitter shift register is empty.
+    (2)       TE - Transmitter FIFO empty: Indicates that the transmitter FIFO is empty.
+    (3)       BR - Break received: Indicates that a BREAK has been received.
+    (4)       OV - Overrun: Indicates that one or more character have been lost due to overrun.
+    (5)       PE - Parity error: Indicates that a parity error was detected.
+    (6)       FE - Framing error: Indicates that a framing error was detected.
+    (7)       TH - Transmitter FIFO half-full: Indicates that the FIFO is less than half-full.
+    (8)       RH - Receiver FIFO half-full: Indicates that at least half of the FIFO is holding data.
+    (9)       TF - Transmitter FIFO full: Indicates that the Transmitter FIFO is full.
+    (10)      RF - Receiver FIFO full: Indicates that the Receiver FIFO is full.
+    (20 - 25) TCNT - Transmitter FIFO count: Shows the number of data frames in the transmitter FIFO.
+    (26 - 31) RCNT - Receiver FIFO count: Shows the number of data frames in the receiver FIFO.
+*/
+typedef enum {APBUART_DR = 0, APBUART_TS, APBUART_TE, APBUART_BR, APBUART_OV, APBUART_PE, APBUART_FE, APBUART_TH, 
```

Comment:
I don't think this is a register definition, this is definition of flags (or flag offsets).

Reply:
ack

Reply:
But not changed...